Enable Tiled Gallery Block on VIP Go

mikeyarce opened the pull request:
#### Changes proposed in this Pull Request:

On VIP Go, we don’t load the Photon module by default because we use our analogous Files Service which works with Tiled Galleries.

This PR adds a check to make sure the Tiled Gallery Block will be available on VIP Go environments.

#### Testing instructions:
- Spin up a VIP Go environment
- Make sure Gutenberg is enabled with `wpcom_vip_load_gutenberg()`
- Verify that the `Tiled Gallery Block` is not present.
- Apply this PR and see it show up!

#### Proposed changelog entry for your changes:
* Add VIP Go Environment check for registering Tiled Gallery Block

mikeyarce committed:
Enable Tiled Gallery Block on VIP Go

On VIP Go, we don’t load the Photon module by default because we use our analogous Files Service which works with Tiled Galleries.

This added check makes sure the Tiled Gallery Block will be available on VIP Go environments.

simison commented:
This will still render these image URLs _with_ Photon instead of serving them via VIPs servers.

Are VIPs file servers running Photon? If so, you could do a change here: 

https://github.com/Automattic/jetpack/blob/e12fef4990a3f32d6e4ae22b6f1258ddb3eb2b86/extensions/blocks/tiled-gallery/utils/index.js#L49

simison commented:
It might as well be valid to just keep those image URLs as wp.com Photon URLs if someone so decides. :-) Just it should be a conscious decision I suppose since VIP sites can generate quite some traffic?

I don't see why the filter couldn't be directly in the file I pointed out, or alternatively, you'd have to turn Photon URLs back to VIP URLs in the backend which seems cumbersome.

Any concerns with that approach @jeherve @mikeyarce ?

jeherve commented:
> Any concerns with that approach @jeherve @mikeyarce ?

I'll let Mikey weigh in on that as I am not really familiar with VIP Go's custom Photon implementation.

mikeyarce committed:
Deliver tiled gallery images through VIP File Service

Adding a check for `is_VIP` that uses a variable passed through `wp_localize_script` to determine if the site is on the VIP plan.

When the site does have a VIP plan, it uses the sites domain to load the images and not the photonized urls.

mikeyarce commented:
@simison Added a check in `isVIP()` and now the images are delivered through the VIP File Service.  

Let me know what you think!

mikeyarce committed:
Remove unneccary parts of `jetpack_plan` array

We don’t need all the data from the  `Jetpack_Plan::get()` function, we just need the `product_slug`.

Fix mispelling.

jeherve reviewed (approved):
This looks good to me! 👍

kraftbj commented:
Looks good.
